src/Plugin/Block/BaristaBlock.php

```diff
@@ -0,0 +1,162 @@
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The cofiguration factory service.
+   * @param \Drupal\Core\Session\AccountProxyInterface $current_user
+   *   The current user service.
+   * @param \Drupal\draco_coffee\DracoCoffeeManager $draco_coffee_manager
+   *   The Draco Coffee Manager service.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The route match.
+   * @param \Drupal\Core\State\StateInterface $state
+   *   The State API service.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $config_factory, AccountProxyInterface $current_user, DracoCoffeeManager $draco_coffee_manager, EntityTypeManagerInterface $entity_type_manager, RouteMatchInterface $route_match, StateInterface $state) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->configFactory = $config_factory;
+    $this->currentUser = $current_user;
+    $this->dracoCoffeeManager = $draco_coffee_manager;
+    $this->entityTypeManager = $entity_type_manager;
+    $this->routeMatch = $route_match;
+    $this->state = $state;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+      $container->get('current_user'),
+      $container->get('draco_coffee.manager'),
+      $container->get('entity_type.manager'),
+      $container->get('current_route_match'),
+      $container->get('state')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function blockAccess(AccountInterface $account) {
+    // Display the block only for users with a particular role.
+    if ($account->hasPermission('administer draco_coffee configuration')) {
+      return AccessResult::allowed()
+        ->addCacheContexts(['route.name'])
+        ->addCacheableDependency($account)
+        ->addCacheTags($this->dracoCoffeeManager->getCacheTags());
+    }
+    return AccessResult::forbidden();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    /** @var AccountProxyInterface $barista */
+    $barista = $this->dracoCoffeeManager->getBarista();
+    if (!$barista) {
+      $markup = $this->t('No coffee being served');
+    }
+    elseif ($barista->id == $this->currentUser->getAccount()->id()) {
+      $markup = $this->t('go make COFFEEEEEEEEEEEEEEEEEEEE!');
+    }
+    else {
+      $markup = $this->t('User @name is making COFFEEEEEEEEEEEEEEEEEEEE', [
+        '@name' => $barista->getAccountName(),
+      ]);
+    }
+
+    $build['content'] = [
+      '#markup' => $markup,
```

Comment:
I think the cacheability metadata should go here.

Reply:
Interesting, then why blockAccess is setting cacheability? This was auto-generated so I thought that it was right. To verify it, I did some research and found https://drupal.stackexchange.com/a/199541/4942 which instead extends `getCacheTags` and `getCacheContexts`.